src/main/java/org/semux/gui/dialog/TransactionDialog.java

```diff
@@ -38,13 +41,23 @@ public TransactionDialog(JFrame parent, Transaction tx) {
         JLabel lblTimestamp = new JLabel(GuiMessages.get("Timestamp") + ":");
         JLabel lblData = new JLabel(GuiMessages.get("Data") + ":");
 
+        JLabel lblGasUsed = new JLabel(GuiMessages.get("GasUsed"));
+        JLabel lblOutput = new JLabel(GuiMessages.get("Output"));
+
+        String notAvailable = GuiMessages.get("NotAvailable");
+
         JTextArea hash = SwingUtil.textAreaWithCopyPopup(Hex.encode0x(tx.getHash()));
         hash.setName("hashText");
         JLabel type = new JLabel(tx.getType().name());
         type.setName("typeText");
         JTextArea from = SwingUtil.textAreaWithCopyPopup(Hex.encode0x(tx.getFrom()));
         from.setName("fromText");
         JTextArea to = SwingUtil.textAreaWithCopyPopup(Hex.encode0x(tx.getTo()));
+
+        // for creates, the TO value should display the contract address
+        if (tx.getType() == TransactionType.CREATE) {
+            to.setText(Hex.encode0x(HashUtil.calcNewAddress(tx.getFrom(), tx.getNonce())));
+        }
```

Comment:
I'd suggest we keep the original value here.

Reply:
Fine with me, but how do UI users find their contract address?  No mechanism exists to find